code/obj/item/rcd.dm

```diff
@@ -690,6 +702,132 @@ Broken RCD + Effects
 				T.req_access = null
 				T.req_access_txt = null
 
+/obj/item/rcd/material
+
+	New()
+		..()
+		src.setMaterial(getMaterial(material_name))
```

Comment:
I don't think the device needs to be made from the material that it creates things from.